calico_upgrade/pkg/commands/dryrun.go

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2016 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/docopt/docopt-go"
+
+	"github.com/projectcalico/calico/calico_upgrade/pkg/clients"
+	"github.com/projectcalico/calico/calico_upgrade/pkg/constants"
+	"github.com/projectcalico/calico/calico_upgrade/pkg/migrate"
+)
+
+func DryRun(args []string) {
+	doc := constants.DatastoreIntro + `Usage:
+  calico-upgrade dry-run
+      [--apiconfigv3=<V3_APICONFIG>]
+      [--apiconfigv1=<V1_APICONFIG>]
+      [--output-dir=<OUTPUTDIR>]
+      [--ignore-v3-data]
+
+Example:
+  calico-upgrade dry-run --apiconfigv3=/path/to/v3/config --apiconfigv1=/path/to/v1/config
+
+Options:
+  -h --help                    Show this screen.
+  --apiconfigv3=<V3_APICONFIG> Path to the file containing connection
```

Comment:
I do not know if I am missing something here but there seems to be a mismatch between the variable name and the description text. On the left, we see "--apiconfigv3=<V3_APICONFIG>". On the right, we see "for the Calico v1 API."

This pattern reverses just below. So, it's quite confusing.

Reply:
Another thing that confuses me a little is which one is the input file and which one is the output. I am thinking that we feed in a v1 manifest and we get back a v3 manifest. If so, that would be a nice thing to include in the descriptions as well, to differentiate the two flags a bit more. In the one flag, you would specify the path where your input v1 file resides. In the other, you would specify where you want the output file to be written.

In addition, should there be any defaults, that would be of great interest as well. For example, I might think that if you don't pass a path, the script might look in the current directory, but that's just a guess.